tests/monitoring_processes/test_monitoring_critical_processes.py

```diff
@@ -0,0 +1,543 @@
+    return expected_alerting_messages
+
+
+def get_num_asics(duthost):
+    """Get number of ASICs on the DUT.
+
+    Args:
+        duthost: Hostname of DUT.
+
+    Returns:
+        An integer which shows number of ASICs on the DUT.
+    """
+    command_num_asics = "python -c 'exec(\"from sonic_py_common import multi_asic\\nprint(multi_asic.get_num_asics())\")'"
+    command_output = duthost.shell(command_num_asics)
+    exit_code = command_output["rc"]
+    pytest_assert(exit_code == 0, "Failed to get the number of ASICs")
+
+    num_asics = command_output["stdout_lines"][0]
+
+    return int(num_asics)
+
+
+def parse_config_entry(config_info):
+    """Parse a single entry of `FEATURE` table.
+
+    Args:
+        config_info: A list which contains the detailed configuration of a
+        container in `FEATURE` table.
+
+    Returns:
+        is_enabled: A string ("enabled|disabled") shows whether this container
+        is enabled or not.
+        has_global_scope: A string ("True|False") shows if a device has multi-ASIC,
+        whether the container should be running in the host.
+        has_per_asic_scope: A string ("True|False") shows if a device has multi-ASIC,
+        whether the container should be running in each ASIC.
+    """
+    is_enabled = ""
+    has_global_scope = ""
+    has_per_asic_scope = ""
+
+    for index, item in enumerate(config_info):
+        if item == "state":
+            is_enabled = config_info[index + 1]
+        elif item == "has_global_scope":
+            has_global_scope = config_info[index + 1]
+        elif item == "has_per_asic_scope":
+            has_per_asic_scope = config_info[index + 1]
+
+    return is_enabled, has_global_scope, has_per_asic_scope
+
+
+def parse_feature_table(duthost, num_asics, skip_containers):
+    """Parses the `FEATURE` table in Config_DB.
+
+    This function will parse the `FEATURE` table in Config_DB to get which containers
+    were enabled and which namespaces these enabled containers reside in.
+
+    Args:
+        duthost: Hostname of DUT.
+        num_asics: An integer shows number of ASICs on the DUT.
+        skip_containers: A list shows which containers will be skipped.
+
+    Returns:
+        A dictionary where keys are container names and values are a list which contains
+        ids of namespaces this container should reside in such as {lldp: ["host", "0", "1"]}
+    """
+    container_list = []
+    containers_in_namespaces = defaultdict(list)
+
+    container_list_command = "redis-cli -n 4 keys \"FEATURE|*\""
+    command_output = duthost.shell(container_list_command)
+    exit_code = command_output["rc"]
+    pytest_assert(exit_code == 0, "Failed to get keys (container names) in `FEATURE` table")
+    for line in command_output["stdout_lines"]:
+        container_list.append(line.split("|")[1].strip())
+
+    for container_name in container_list:
+        if container_name in skip_containers:
```

Comment:
why not remove the skip list from working list before the loop?

Reply:
Fixed.